app/settings/tokens/-components/edit-form/component.ts

```diff
@@ -66,12 +66,18 @@ export default class TokenForm extends Component {
 
     @action
     async deleteToken() {
-        try {
-            await this.token!.destroyRecord();
-            this.toast.success(this.intl.t('settings.tokens.deleted'));
-            this.router.transitionTo('settings.tokens');
-        } catch {
-            this.toast.error(this.intl.t('settings.tokens.delete-error'));
+        if (this.token) {
+            try {
+                await this.changeset.validate();
+                if (this.changeset.isValid) {
+                    this.token.deleteRecord();
+                    this.token.save();
+                    this.toast.success(this.intl.t('settings.tokens.deleted'));
```

Comment:
The `save` might not be successful when you toast the success message because there is no `await` on the promise.

Reply:
Ok.

Reply:
An await has now been added to the save and yes, it now works fully on the first try.